rfcs/text/0014_scheduled_reports_new_plugin.md

```diff
@@ -0,0 +1,164 @@
+
+- Rendering characteristics of a page (Dashboards, for instance) should be maintained and owned by that team.
+- The general pipeline of reporting: queueing, export execution, browser binaries, persistance of assets, and the REST APIs are owned by the reporting team.
+- Other characteristics, such as licensing and role's, can be established in one (or both) places.
+
+At its core, we want to expand upon the features and fixes that can make reporting a killer-feature of Kibana. In order to do so, it's time to shed the maintenance cost and upkeep of application-specific logic to the underlying team's themselves.
+
+# Overall design
+
+The new plugin would follow Kibana's recommended and best practices: establishing a TypeScript-based plugin that others can add as a dependency and consume. The following is a brief overview of how an application would incorporate reporting into their application stack.
+
+## 1. Add a dependency on reporting
+In order to use reporting, a plugin simply has to add a dependency on reporting in their `kibana.json`:
+
+```json
+{
+  "id": "super-dashboards",
+  "version": "8.0.0",
+  "kibanaVersion": "kibana",
+  "optionalPlugins": ["reporting"],
+  "server": true,
+  "ui": true
+}
+```
+
+Now that `reporting` is added as a plugin, downstream plugins can now integrate reporting into their stack.
+
+## 2. Using the plugin
+
+Similar to other plugins, reporting will offer a module that allows consumers to define the types and kinds of reports allowed by the page. For example, the below call will register a new "PDF" report to the page:
+
+```js
+// plugin.ts server
+public setup(core: CoreSetup, plugins: PluginSetup) {
+  const { reporting } = plugins;
+
+  reporting.registerReport({
+    id: Symbol('super-dashboard-report'),
+    tags: ['super-dashboard', 'triple-word-score', 'synergy'],
+    type: 'pdf',
+    label: 'Export Dashboard',
+    visibility: 'user',
+    intervals: ['immediate', 'monthly', 'weekly'],
+    connectors: ['email', 'slack'],
+    before: (req, res, context): Promise<boolean> => {},
+    after: (result: Buffer) => void,
+  });
+}
+
+// plugin.ts public
+public start(core: CoreStart, plugins: PluginStart) {
+  const { reporting } = plugins;
+  const dashboardReports = reporting.findReportById(Symbol('super-dashboard-report'));
+}
+
+// Later, in some React component:
+<Button onClick={() => dashboardReports.open(`http://localhost:5601/my-dope-dashboard?pdf-export=true`)}>
+```
+
+Care should be taken, in the final API design, to strike a good balance between what reporting takes ownership of, and the page/application itself. For instance, adding the Button to the page is likely something that application owners will want to control, but not re-creating the entire scheduling interface. Once consensus has been established on the overall design we'll revisit this handover and devise what belongs where.
```

Comment:
we should probably come up with the final API design before starting down this route as this seems to be the most important aspect of this RFC